polkadot/node/service/src/chain_spec.rs

```diff
@@ -682,7 +402,7 @@ pub fn rococo_staging_testnet_config() -> Result<RococoChainSpec, String> {
 	.with_name("Rococo Staging Testnet")
 	.with_id("rococo_staging_testnet")
 	.with_chain_type(ChainType::Live)
-	.with_genesis_config_patch(rococo_staging_testnet_config_genesis())
+	.with_genesis_config_preset_name("staging_testnet")
```

Comment:
We need flexibility here. We cannot predict all genesis config presets and put them in enum. We also need to allow runtime devs to define their own presets (e.g. for their own testnets or CI tests).

Reply:
It could have `Benchmarking` and `Named(RuntimeString)` variants? But i dont want to prevent this MR from merging, just ideas.

Reply:
> How would the benchmark-runner know what preset to pick?

How is it done today?

Reply:
It uses the default. But that meant that i had to change the default for a few pallets to be the worst cast.  
Not sure what is best - either to have a dedicated preset - or to add more traits to the pallets to overwrite the config as preparation for benchmarks.   
For now i can probably continue to use the default, as it seems to work okay-ish.

Reply:
> It could have `Benchmarking` and `Named(RuntimeString)` variants? But i dont want to prevent this MR from merging, just ideas.

It is inline with my previous proposal. Maybe something to consider in the future:
https://github.com/paritytech/polkadot-sdk/pull/2714#issuecomment-2015125305